cache/cache.go

```diff
@@ -20,20 +21,37 @@ type Cache struct {
 	cacheFile   string
 	allDirty    bool
 	dirtyFiles  []string
+
+	opts []shell.RunOption
 }
 
-func New() *Cache {
+func New(opts ...shell.RunOption) *Cache {
 	return &Cache{
-		ranOnce:   make(map[*taskrunner.Task]bool),
-		cacheFile: path.Join(CacheDir, "example.json"),
+		ranOnce: make(map[*taskrunner.Task]bool),
+		opts:    opts,
 	}
 }
 
-func (c *Cache) Start(opt shell.RunOption) error {
+func (c *Cache) Option(r *taskrunner.RunOptions) {
+	r.ReporterFns = append(r.ReporterFns, func(ctx context.Context, executor *taskrunner.Executor) error {
+		c.cacheFile = getCacheFilePath(executor.Config().WorkingDir())
+		c.Start()
+		<-ctx.Done()
+		c.Finish()
+		return nil
+	})
+}
+
+func getCacheFilePath(dir string) string {
+	hashedName := strings.Replace(dir, "/", "%", -1)
```

Comment:
how about replacing any non-alphanum with `-`?

Reply:
```
regexp.MustCompile(`[^A-Za-z\-_]`)
```

Reply:
Could do that - I figured slashes would be the only dangerous thing though. NSO